scipy/linalg/_decomp_qr.py

```diff
@@ -117,6 +123,35 @@ def qr(a, overwrite_a=False, lwork=None, mode='full', pivoting=False,
     ((9, 6), (6, 6), (6,))
 
     """
+    xp = array_namespace(a)
+    if is_numpy(xp):
+        return _qr(a, overwrite_a=overwrite_a, lwork=lwork, mode=mode,
+                   pivoting=pivoting, check_finite=check_finite)
+    if overwrite_a:
+        raise ValueError(arg_err_msg("overwrite_a"))
+    if lwork is not None:
+        raise ValueError(arg_err_msg("lwork"))
+    if mode not in {'full', 'economic', 'reduced', 'complete'}:
+        raise ValueError(...)
+    if pivoting:
+        raise ValueError(arg_err_msg("pivoting"))
+    if not check_finite:
+        raise ValueError(arg_err_msg("check_finite"))
+    if hasattr(xp, 'linalg'):
+        # PROBLEM: scipy default mode is full (complete)
+        # but API default is reduced (economic).
+        ...
```

Comment:
Leave the default as full/complete for now. Can note this discrepancy with the array API in the main PR.

Reply:
This has been addressed.